lightning/src/offers/invoice_request.rs

```diff
@@ -528,9 +529,12 @@ impl UnsignedInvoiceRequest {
 		let mut bytes = Vec::new();
 		unsigned_tlv_stream.write(&mut bytes).unwrap();
 
-		let tagged_hash = TaggedHash::from_valid_tlv_stream_bytes(SIGNATURE_TAG, &bytes);
+		let experimental_bytes = Vec::new();
+
+		let tlv_stream = TlvStream::new(&bytes).chain(TlvStream::new(&experimental_bytes));
```

Comment:
At this point, we will fail parsing any `Offer` with TLVs in the experimental range. So `offer_bytes` will not contain any experimental TLVs.

It seemed easier to reason about by adding `experimental_bytes` in a dedicated commit. Otherwise, the change gets lost in the parsing commits (e.g., 927c4776a8b9f917a03015ea9665c204cad63d35), which are large changesets.

Also note that we never use parts from `as_tlv_stream` that don't originated from us. We need to use the bytes instead because they may contain unknown odd TLVs.

Reply:
Added another fixup for `UnsignedStaticInvoice`.